integration_tests/sdk/multiple_output_test.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+from aqueduct.error import AqueductError
+from utils import run_flow_test
+
+from aqueduct import op
+
+
+def test_multiple_outputs(client):
+    @op(num_outputs=2)
+    def generate_two_outputs():
+        return "hello", 1234
+
+    @op
+    def append_to_str(input_str):
+        return input_str + " world."
+
+    @op
+    def double_number(num):
+        return 2 * num
+
+    str_artifact, int_artifact = generate_two_outputs()
+    assert str_artifact.get() == "hello"
+    assert int_artifact.get() == 1234
+
+    str_output = append_to_str(str_artifact)
+    int_output = double_number(int_artifact)
+    assert str_output.get() == "hello world."
+    assert int_output.get() == 2468
+
+    run_flow_test(client, artifacts=[str_output, int_output], delete_flow_after=False)
+
+
+def test_multiple_outputs_user_failure(client):
+    @op(num_outputs=3)
+    def generate_two_outputs():
+        return "hello", 1234
+
+    with pytest.raises(AqueductError):
```

Comment:
I think based on the implementation this should raise a `InvalidUserArgumentException`? Not sure why it worked.

Reply:
It should be failing at preview execution, which leads to an AqueductError. Where in the code did you expect `InvalidUserArgumentException` to be thrown? We can't check the number of outputs until we actually run the code on the backend.

Reply:
Oh I misread the code. Ignore this.